### ISSUE TITLE
Create demo project and ensure completion of accelerometer/gyro/magnetometer on both iOS and Android

### ISSUE THREAD

**Issue by BastiaanOlij** [godotengine/godot#7531]
This is more a reminder then an issue, Akien asked me to raise this so we don't forget to do this.

PR #7127 has now been merged into the Godot master to enable this functionality on iOS. This worked fine for Godot 2 but we're waiting on completion of changes to the new renderer engine before this can be properly tested in Godot 3.

On Android #7503 needs to be finished to ensure compatibility between the two platforms. I unfortunately do not have access to Android.

Last but not least, it would be good to create one or two demo projects both to include in the Godot demos and to test this functionality to ensure it works consistently.

I do have an early prototype using this functionality to control the camera for a little Cardboard VR game but there are a few other simple concepts that would be good to demo such as maybe a compass or the basis or an AR type game (I hope to look into capturing camera input soon).

**Comment by akien-mga** [comment]
And if someone is up to it, once iOS and Android are 100% in sync regarding this API, it might be worth checking if UWP could have it too for Windows phones.

**Comment by BastiaanOlij** [comment]
I didn't realise we had OpenGL support on UWP, I thought Microsoft had pretty much forced DirectX on that platform...

Hmmm, Godot on Hololens.....? This could be interesting :)

**Comment by vnen** [comment]
BTW, it's not quite "OpenGL support". We use ANGLE to translate OpenGL API to DirectX.

**Comment by BastiaanOlij** [comment]
@vnen cool, I know very little about the UWP platform as I do most of my development on Mac OS X. I'm getting very disillusioned with Microsoft because they seem to really be bloating their development tools and approach. I wanted to play around with the Hololens SDK only to find out that even on my brand new HP laptop with top end Radion graphics card the SDK simply won't run because the virtual machine emulator just gets stuck. It's probably lack of experience on my part but...

Anyway, for the Hololens too I thought it was DirectX or the highway but when I realised Godot ran on UWP I googled around and found out about ANGLE and even a couple of people using ANGLE to build an OpenGL based AR engine that amongst others is deployable on Hololens.

With the new affordable Lenovo headset announced I'm getting very interested in this platform :) It be interesting to tackle that some day for Godot

**Comment by reduz** [comment]
I'm reminding you of your reminder, but kicking your reminder to 3.1

**Comment by BastiaanOlij** [comment]
@reduz I'll think of something. Note that on Android the axis are still reversed (#7503), someone needs to look into that and see how and if device orientation has any effect on this. I can't make the change lacking a capable android phone.

Else we'll have to add android checks to any demo project. But I'll find some time to add a nice demo :)

**Comment by BastiaanOlij** [comment]
I created this:
https://github.com/godotengine/godot-demo-projects/pull/90

Improvements are needed on handling the magnetometer data but I think we can close this.
